python/lsst/analysis/tools/interfaces/datastore/_dispatcher.py

```diff
@@ -232,19 +236,25 @@ def _create_topic(self, topic_name: str) -> bool:
             "replication_factor": REPLICATION_FACTOR,
         }
 
-        r = requests.post(
-            f"{self.url}/v3/clusters/{self.clusterId}/topics", json=topic_config, headers=headers
-        )
-
-        if r.status_code == requests.codes.created:
+        with http_client() as session:
+            r = session.post(
+                f"{self.url}/v3/clusters/{self.clusterId}/topics", json=topic_config, headers=headers
+            )
+        try:
+            r.raise_for_status()
             log.debug("Created topic %s.%s", self.namespace, topic_name)
             return True
-        elif r.status_code == requests.codes.bad_request:
-            log.debug("Topic %s.%s already exists.", self.namespace, topic_name)
-            return True
-        else:
-            log.error("Uknown error occured creating kafka topic %s %s", r.status_code, r.json())
```

Comment:
This change specifically avoids calling a `.json()` method on a response that may not be json-encoded. We don't have a contract with the called service so we use the standard `reason` string instead.